core/ledger/kvledger/upgrade_dbs.go

```diff
@@ -24,29 +32,39 @@ func UpgradeDBs(rootFSPath string) error {
 	defer fileLock.Unlock()
 
 	logger.Infof("Ledger data folder from config = [%s]", rootFSPath)
+	dbPath := LedgerProviderPath(rootFSPath)
+	db := leveldbhelper.CreateDB(&leveldbhelper.Conf{DBPath: dbPath})
+	db.Open()
+	defer db.Close()
 
-	// upgrades idStore (ledgerProvider) and drop databases
-	if err := UpgradeIDStoreFormat(rootFSPath); err != nil {
+	idStore := &idStore{db, dbPath}
+	format, err := idStore.getFormat()
+	if err != nil {
 		return err
 	}
+	if bytes.Equal(format, []byte(dataformat.Version20)) {
+		logger.Info("Ledger data format is current, do nothing.")
+		return nil
+	}
```

Comment:
I recall that we purposely return an error if the idStore format is neither nil (1.x) or 2.0 with a forward thinking. When 3.0 if available, the method will only upgrade from 2.0 to 3.0 and return an error for other formats. Given it, idStore.hasLastestFormat() may not be helpful if there are more than 2 formats. Of course, we can simplify the code for now since we only have 1.x and 2.0.
@manish-sethi what do you think?

Reply:
I am not questioning the return of an `error`. I mentioned the side-effect in upgradeFormat and lines 41 to 48 occur twice overall. The `updateFormat()` is in fact doing `updateFormatIfCurrentFormatIsNotV20()`. If we can split that into two, we can avoid this code duplication as well as the side-effect.